src/ashell/jerry-code.c

```diff
@@ -221,9 +221,14 @@ static bool add_requires(requires_list_t **list, char *filebuf)
                 // Check that this is a *.js require and not a zephyr require
                 // Also check that the file hasn't already been loaded
                 ext_check = &filestr[len - 3];
-
                 if (strcmp(ext_check, ".js") == 0 && !list_contains(list, filestr)) {
-                    add_to_list(list, filestr);
+                    if (fs_valid_filename(filestr)) {
+                        add_to_list(list, filestr);
+                    }
+                    else {
```

Comment:
you can just in else statement to free the filestr, and return false immediately, instead of adding this valid bool check, it's cleaner